compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2519,24 +2548,32 @@ genLoadAddressConstant(TR::CodeGenerator * cg, TR::Node * node, uintptr_t value,
          comp->getStaticPICSites()->push_front(cursor);
          isCompressedClassPointer = comp->useCompressedPointers();
          }
-      assumePatch = true;
+
+      if (comp->getOption(TR_EnableHCR) && node->getOpCode().hasSymbolReference() &&
+          node->getSymbol()->isStatic() && node->getSymbol()->isClassObject())
+         {
+         comp->getStaticHCRPICSites()->push_front(cursor);
+         }
+
+      TR_ASSERT(!isCompressedClassPointer || ((value & CONSTANT64(0xFFFFFFFF00000000)) == 0), "Compressed class pointers are assumed to fit in 32 bits");
+      // IIHF is only needed when addresses do not fit into 32 bits
```

Comment:
You might want to update this comment to mention LARL - i.e., "when addresses do not fit into 32 bits, and LARL could not be used"